libvast/builtins/formats/json.cpp

```diff
@@ -0,0 +1,356 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/adaptive_table_slice_builder.hpp>
+#include <vast/arrow_table_slice.hpp>
+#include <vast/concept/parseable/numeric/bool.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/concept/printable/vast/json.hpp>
+#include <vast/defaults.hpp>
+#include <vast/detail/assert.hpp>
+#include <vast/generator.hpp>
+#include <vast/operator_control_plane.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/record_batch.h>
+
+#include <simdjson.h>
+
+namespace vast::plugins::json {
+
+namespace {
+
+using parse_result = std::optional<simdjson::error_code>;
+
+class parser_buffer {
```

Comment:
Can you move this class to `vast/detail/padded_buffer.hpp`, rename it to `padded_buffer` and take the padding size as a template argument? I think this is more generally useful than just here, and it makes sense to have it in its own place.

NB: I also think that the entire class can be made `constexpr`, which would guarantee that it works correctly.

Reply:
Done. I was wondering if we want to make it more generic as it's in a common header file.
I ended up thinking that is not worth the time for now. If someone were to use it with different allocation strategy or a different value type than 'char' then he can refactor it